starknet-client/integration/main.rs

```diff
@@ -0,0 +1,128 @@
+//! Defines integration tests for
+#![deny(unsafe_code)]
+#![deny(missing_docs)]
+#![deny(clippy::missing_docs_in_private_items)]
+#![allow(incomplete_features)]
+#![feature(generic_const_exprs)]
+
+use clap::Parser;
+use common::types::chain_id::ChainId;
+use eyre::Result;
+use mpc_stark::algebra::scalar::Scalar;
+use rand::thread_rng;
+use starknet_client::client::{StarknetClient, StarknetClientConfig};
+use test_helpers::{
+    assert_true, contracts::parse_addr_from_deployments_file, integration_test_async,
+    integration_test_main,
+};
+use tracing::log::LevelFilter;
+
+/// The hostport that the test expects a local devnet node to be running on
+///
+/// This assumes that the integration tests are running in a docker-compose setup
+/// with a DNS alias `sequencer` pointing to a devnet node running in a sister container
+const DEVNET_HOSTPORT: &str = "http://sequencer:5050";
+
+/// The arguments used to run the integration tests
+#[derive(Debug, Clone, Parser)]
+#[command(author, version, about, long_about=None)]
+struct CliArgs {
+    /// The private key to use for the Starknet account during testing
+    ///
+    /// Defaults to the first pre-deployed account on `starknet-devnet` when
+    /// run with seed 0
+    #[arg(
+        short = 'p',
+        long,
+        default_value = "0x300001800000000300000180000000000030000000000003006001800006600"
+    )]
+    starknet_pkey: String,
+    /// The address of the account contract to use for testing
+    ///
+    /// Defaults to the first pre-deployed account on `starknet-devnet` when
+    /// run with seed 0
+    #[arg(
+        long,
+        default_value = "0x3ee9e18edc71a6df30ac3aca2e0b02a198fbce19b7480a63a0d71cbd76652e0"
+    )]
+    starknet_account_addr: String,
+    /// The address of the darkpool deployed on Starknet at the time the test is started
+    ///
+    /// If not provided, the test will deploy a new darkpool contract
+    #[arg(long)]
+    darkpool_addr: Option<String>,
+    /// The location of a `deployments.json` file that contains the addresses of the
+    /// deployed contracts
+    #[arg(long)]
+    deployments_path: Option<String>,
```

Comment:
do we need both of these arguments? i would assume that if the darkpool is deployed, it would be included in `deployments.json`

Reply:
i see below that we expect one of the two of these to be set.
1) we could use the clap `group` attribute to ensure exactly one is set, if this is the desired behavior
2) this is confusing to me because the doc comment here seems to imply that its okay for both of these flags to be unset, as if this is the case then the test will deploy the darkpool contract

Reply:
I'll use clap group and update the doc comment,  copied these args from old tests